test/unit/tag_test.rb

```diff
@@ -92,4 +92,30 @@ class TagTest < ActiveSupport::TestCase
     assert !node(:blog).responses('replication').empty?
     assert node(:blog).response_count('replication') > 0
   end
+
+  test "returns empty array if users are  following both the given tags and this tag" do
+    tag = tags(:spam)
+    given_tags = [tags(:chapter)]
+    assert_equal [], tag.followers_who_dont_follow_tags(given_tags)
+  end
+
+  test " returns users following this tags but not given tags" do
+    test = tags(:test)       # users following tag are bob, unbanned_spammer, admin, and following: false for jeff
+    awesome = tags(:awesome) # users following tag1 are bob, unbanned_spammer, moderator
+    spam = tags(:spam)       # users following tag2 are spammer, newcomer, and following: false for unbanned_spammer
+    given_tags = [awesome, spam]
+    assert_equal [rusers(:admin)], test.followers_who_dont_follow_tags(given_tags)
+    # now make unbanned_spammer following: false for both 'awesome' and 'spam' tags:
+    tag_selection(:selection_four).update_attribute('following', false)
+    given_tags = [awesome, spam]
+    assert_equal [rusers(:admin), rusers(:unbanned_spammer)], test.followers_who_dont_follow_tags(given_tags)
```

Comment:
OK, i've added a commit to try to demonstrate how following = false should affect these queries. See? Hopefully if I've gotten it right, this should pass.